### DISCUSSION TITLE
What parts of the hardware have to be modified to support AI Deck, Loco Deck, and Flow Deck simultaneously?

### DISCUSSION THREAD

**Discussion by 787-10** [bitcraze/discussions#856] [Q&A]
Hello Bitcraze community, this is a continuation of the previous issue that I have posted [here](https://github.com/orgs/bitcraze/discussions/854). I spent today testing with only the AI deck and the loco deck, and I believe that GPIO-1 seems to be the conflict pin (that is being used by both the AI Deck and the Loco Deck simultaneously).

To try to see if I could force a switch for the Loco Deck chip select to use GPIO 2, I laid out the drone and the loco deck separately (with the AI deck attached on the drone), and I manually connected the corresponding pins (with GPIO1 enabled during the first test and GPIO2 enabled during the second test) and updated the firmware accordingly as shown in the attached diagram. 

<img width="1131" alt="Screenshot 2023-08-04 at 12 34 08 AM" src="https://github.com/bitcraze/discussions/assets/60076597/90f278de-d01d-4222-b586-f608b6bc076f">

The corresponding pins were determined from [this page](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#deck-pin-allocation) along with the deck pin allocation table; the pins that I connected were all of the table entries for the Loco Deck that were not blank (so I believed that those were the pins in use by the loco deck only).

However, the loco deck was not being detected during either of those times. At the same time, I came across discussion posts #563 and #776. The first one seemed to indicate that compatibility can be forced via a firmware update; however, it seems that I would have to solder the bridges for GS3 and GS4 across those rectangles while cutting off the GS1-GS2 connection (I am still a bit unclear as to what that means). Given that I am planning on using the flow deck as well, which means that GPIO3 is the chip select for the flow deck, what specific hardware modifications will need to be made to the loco deck? Additionally, what specific firmware updates will need to be made as well?

**Comment by tobbeanton** [comment]
Please check our other post. It was a lot easier to fix then I thought. No hardware modification needed.
